src/compiler/core.ts

```diff
@@ -1186,6 +1186,12 @@ namespace ts {
         return path.substr(0, Math.max(getRootLength(path), path.lastIndexOf(directorySeparator)));
     }
 
+    function getBasename(path: Path): Path;
+    function getBasename(path: string): string;
+    function getBasename(path: string): any {

```

Comment:
You shouldn't need to use `any` - our overload compatibility rules are bidirectional for return types.